### PR TITLE
[DO NOT MERGE] feat: add affirm payment method

### PR DESCRIPTION
**Pull request by jameswoo-stripe**
## Summary

- Add Affirm payment method bindings
- Add Affirm payment method to example app
- Update example server to handle item ids and prices, this was needed because affirm requires a minimum of $USD 50

iOS

https://user-images.githubusercontent.com/99316447/155247122-a023d719-3523-4c76-b208-899853629c08.mov

Android

https://user-images.githubusercontent.com/99316447/155246905-7d7c30cb-ea2d-4389-be23-6047e815c738.mov

## Testing

- [x] I tested this manually
- [ ] I added automated tests

## Documentation

Select one: 
- [ ] I have added relevant documentation for my changes.
- [ ] This PR does not result in any developer-facing changes.

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -123,6 +123,8 @@ repositories {
                     "Ensure you have you installed React Native as a dependency in your project and try again."
     )
   }
+
+  maven { url 'https://jitpack.io' }
```
Comment (jameswoo-stripe) [inline android/build.gradle:127]: This will be removed and the gradle dependency will be corrected

File: example/android/app/build.gradle [hunk 2]
```diff
@@ -204,7 +204,7 @@ dependencies {
     implementation "com.facebook.react:react-native:+"  // From node_modules
     implementation "androidx.swiperefreshlayout:swiperefreshlayout:1.0.0"
     implementation 'com.google.android.material:material:1.4.0'
-    implementation 'com.stripe:stripe-wechatpay:17.1.0'
+    // implementation 'com.stripe:stripe-wechatpay:17.1.0'
```
Comment (jameswoo-stripe) [inline example/android/app/build.gradle:207]: This change will be removed, just need this temporarily to build correctly if testing locally.

File: example/ios/Podfile [hunk 3]
```diff
@@ -10,6 +10,9 @@ target 'StripeSdkExample' do
     # to enable hermes on iOS, change `false` to `true` and then install pods
     hermes_enabled: false
   )
+  pod 'Stripe', :git => 'https://github.com/stripe-ios/stripe-ios.git', :branch => 'private'
+  pod 'StripeUICore', :git => 'https://github.com/stripe-ios/stripe-ios.git', :branch => 'private'
+  pod 'StripeCore', :git => 'https://github.com/stripe-ios/stripe-ios.git', :branch => 'private'
```
Comment (jameswoo-stripe) [inline example/ios/Podfile:15]: This will be removed, used for testing locally

**Review by charliecruzan-stripe — CHANGES_REQUESTED** [review]
this is really great 🎉 just a couple nits is all from me. 

As for testing- for some reason, the Afterpay test was disabled before I got here, my guess is it's just flaky, but let's try adding one for affirm. It can be done in the same file right below [the afterpay test](https://github.com/stripe/stripe-react-native/blob/master/e2e/paymentsWithRedirects.test.ts#L156-L164), and it'll look super similar

**Comment by charliecruzan-stripe** [comment]
replaced by https://github.com/stripe/stripe-react-native/pull/1036

**Pull request closed without merging**